mkosi/__init__.py

```diff
@@ -2170,6 +2173,14 @@ def run_verb(args: MkosiArgs, presets: Sequence[MkosiConfig]) -> None:
 
     last = presets[-1]
 
+    if args.verb == Verb.download:
+        workspace = tempfile.TemporaryDirectory(dir=last.workspace_dir, prefix=".mkosi-tmp")
+        with workspace, scopedenv({"TMPDIR" : workspace.name}):
```

Comment:
Also this `with` statement is duplicated here and in `build_images()`. Maybe we should opt for a function with the context manager decorator for this (assuming this is even the correct way to go)?